core/blockchain.go

```diff
@@ -1448,56 +1500,59 @@ func (self *BlockChain) WriteBlock(block *types.Block) (status WriteStatus, err
 
 	// If TDs are the same, randomize.
 	if tdCompare == 0 {
-		// Reduces the vulnerability to selfish mining.
+		// Reduces the vulnerability to bcish mining.
 		// Please refer to http://www.cs.cornell.edu/~ie53/publications/btcProcFC.pdf
-		reorg = mrand.Float64() < 0.5
+		// Split same-difficulty blocks by number, then at random
+		reorg = block.NumberU64() < bc.currentBlock.NumberU64() || (block.NumberU64() == bc.currentBlock.NumberU64() && mrand.Float64() < 0.5)
```

Comment:
This is a notice-worthy change, but common sensey-y. Before flipping a coin in case TDs are same, opt-in if the incoming block number is below ours.